### ISSUE TITLE
Not rewriting paths

### ISSUE THREAD

**Issue by brandonkal** [dimadin/commonWP#2]
I really like the idea of this plugin but it is not doing the rewrites for core or plugin files. After making some page requests on the site and giving the background process some time, a
`wp commonwp paths list active` shows that hardly anything is rewritten:

Note: I have removed theme files from the list below, as my theme is not public and I do not expect those paths to be rewritten. However, the paths below should all be rewritten. The plugin is network activated.

Paths that can be rewritten (active):

| Relative Path                             | Remote Path            | Timeout   |
| ----------------------------------------- | ---------------------- | --------- |
| https://s.w.org/images/core/emoji/11/svg/ | /npm/twemoji@11/2/svg/ | in 7 days |
| https://s.w.org/images/core/emoji/2/svg/  | /npm/twemoji@2/2/svg/  | in 7 days |

Paths that are not rewritten (inactive):

| Relative Path                                                                                 | Timeout     |
| --------------------------------------------------------------------------------------------- | ----------- |
| #SITE#/wp-includes/css/dashicons.min.css?ver=4.9.8                                            | in 24 hours |
| #CONTENT#/plugins/query-monitor/assets/query-monitor.css?ver=1537335244                       | in 24 hours |
| #CONTENT#/plugins/debug-bar-plugin-activation/css/debug-bar-plugin-activation.min.css?ver=1.0 | in 24 hours |
| #CONTENT#/plugins/query-monitor/assets/query-monitor.js?ver=1531873320                        | in 24 hours |
| #SITE#/wp-includes/js/jquery/ui/effect.min.js?ver=1.11.4                                      | in 24 hours |
| #SITE#/wp-includes/js/jquery/ui/effect-slide.min.js?ver=1.11.4                                | in 24 hours |
| #CONTENT#/plugins/debug-bar-plugin-activation/js/debug-bar-plugin-activation.min.js?ver=1.0   | in 24 hours |
| #SITE#/wp-includes/js/wp-embed.min.js?ver=4.9.8                                               | in 24 hours |
| #CONTENT#/plugins/query-monitor/assets/query-monitor.js                                       | in 24 hours |
| #CONTENT#/plugins/debug-bar-plugin-activation/js/debug-bar-plugin-activation.min.js           | in 24 hours |
| #CONTENT#/plugins/query-monitor/assets/query-monitor.css                                      | in 24 hours |
| #CONTENT#/plugins/debug-bar-plugin-activation/css/debug-bar-plugin-activation.min.css         | in 24 hours |
| #SITE#/wp-includes/css/admin-bar.min.css?ver=4.9.8                                            | in 24 hours |
| #SITE#/wp-includes/js/thickbox/thickbox.css?ver=4.9.8                                         | in 24 hours |
| #SITE#/wp-admin/css/common.min.css?ver=4.9.8                                                  | in 24 hours |
| #SITE#/wp-admin/css/forms.min.css?ver=4.9.8                                                   | in 24 hours |
| #SITE#/wp-admin/css/admin-menu.min.css?ver=4.9.8                                              | in 24 hours |
| #SITE#/wp-admin/css/dashboard.min.css?ver=4.9.8                                               | in 24 hours |
| #SITE#/wp-admin/css/list-tables.min.css?ver=4.9.8                                             | in 24 hours |
| #SITE#/wp-admin/css/edit.min.css?ver=4.9.8                                                    | in 24 hours |
| #SITE#/wp-admin/css/revisions.min.css?ver=4.9.8                                               | in 24 hours |
| #SITE#/wp-admin/css/media.min.css?ver=4.9.8                                                   | in 24 hours |
| #SITE#/wp-admin/css/themes.min.css?ver=4.9.8                                                  | in 24 hours |
| #SITE#/wp-admin/css/about.min.css?ver=4.9.8                                                   | in 24 hours |
| #SITE#/wp-admin/css/nav-menus.min.css?ver=4.9.8                                               | in 24 hours |
| #SITE#/wp-includes/css/wp-pointer.min.css?ver=4.9.8                                           | in 24 hours |
| #SITE#/wp-admin/css/widgets.min.css?ver=4.9.8                                                 | in 24 hours |
| #SITE#/wp-admin/css/site-icon.min.css?ver=4.9.8                                               | in 24 hours |
| #SITE#/wp-admin/css/l10n.min.css?ver=4.9.8                                                    | in 24 hours |
| #SITE#/wp-includes/css/buttons.min.css?ver=4.9.8                                              | in 24 hours |
| #SITE#/wp-admin/css/ie.min.css?ver=4.9.8                                                      | in 24 hours |
| #SITE#/wp-includes/css/wp-auth-check.min.css?ver=4.9.8                                        | in 24 hours |
| #CONTENT#/plugins/perfmatters/css/style.css?ver=1.2.5                                         | in 24 hours |
| er=1.3.1                                                                                      |             |
| #SITE#/wp-includes/js/jquery/jquery.js?ver=1.12.4                                             | in 24 hours |
| #SITE#/wp-includes/js/jquery/jquery-migrate.min.js?ver=1.4.1                                  | in 24 hours |
| #SITE#/wp-includes/js/utils.min.js?ver=4.9.8                                                  | in 24 hours |
| #SITE#/wp-includes/js/json2.min.js?ver=2015-05-03                                             | in 24 hours |
| #SITE#/wp-includes/js/hoverIntent.min.js?ver=1.8.1                                            | in 24 hours |
| #SITE#/wp-admin/js/common.min.js?ver=4.9.8                                                    | in 24 hours |
| #SITE#/wp-includes/js/admin-bar.min.js?ver=4.9.8                                              | in 24 hours |
| #SITE#/wp-includes/js/thickbox/thickbox.js?ver=3.1-20121105                                   | in 24 hours |
| #SITE#/wp-includes/js/underscore.min.js?ver=1.8.3                                             | in 24 hours |
| #SITE#/wp-includes/js/backbone.min.js?ver=1.2.3                                               | in 24 hours |
| #SITE#/wp-includes/js/wp-util.min.js?ver=4.9.8                                                | in 24 hours |
| #SITE#/wp-includes/js/wp-backbone.min.js?ver=4.9.8                                            | in 24 hours |
| #SITE#/wp-includes/js/wp-a11y.min.js?ver=4.9.8                                                | in 24 hours |
| #SITE#/wp-includes/js/customize-base.min.js?ver=4.9.8                                         | in 24 hours |
| #SITE#/wp-admin/js/theme.min.js?ver=4.9.8                                                     | in 24 hours |
| #SITE#/wp-admin/js/updates.min.js?ver=4.9.8                                                   | in 24 hours |
| #SITE#/wp-admin/js/svg-painter.js?ver=4.9.8                                                   | in 24 hours |
| #SITE#/wp-includes/js/heartbeat.min.js?ver=4.9.8                                              | in 24 hours |
| #SITE#/wp-includes/js/wp-auth-check.min.js?ver=4.9.8                                          | in 24 hours |
| heckbox.js?ver=1.3.1                                                                          |             |
| heckbox.js                                                                                    |             |
| #CONTENT#/plugins/perfmatters/css/style.css                                                   | in 24 hours |
| #SITE#/wp-includes/js/jquery/ui/core.min.js?ver=1.11.4                                        | in 24 hours |
| #SITE#/wp-admin/js/plugin-install.min.js?ver=4.9.8                                            | in 24 hours |
| #SITE#/wp-includes/js/zxcvbn-async.min.js?ver=1.0                                             | in 24 hours |
| #SITE#/wp-admin/js/password-strength-meter.min.js?ver=4.9.8                                   | in 24 hours |

There are no stored paths to process (queued).

**Comment by brandonkal** [comment]
It appears that the issue is occurring here on line 705:
https://github.com/dimadin/commonWP/blob/06a2bc4020342990f03ad5f35ce75be46f25adf0/inc/Process.php#L700-L705

Would it not be better to read the local file directly? The issue is that wp_safe_remote_get fails when using development SSL certificates. So the request to https://site.test fails as it is not verified by the included CAs.

**Comment by dimadin** [comment]
Hey Brandon, thanks for report and for investigating what might be the issue! It's great to see that someone else is using WP-CLI commands. 👍

> Would it not be better to read the local file directly?

During development, I found that to be unreliable because of non-default configurations.

> The issue is that wp_safe_remote_get fails when using development SSL certificates.

[`wp_safe_remote_get()`](https://developer.wordpress.org/reference/functions/wp_safe_remote_get/) is used to prevent redirections. If there is a problem with SSL certificate, this is because [default setting](https://developer.wordpress.org/reference/classes/WP_Http/request/#parameters) for WP HTTP requests is to verify SSL certificates. That setting can be filtered, here is the filter that would prevent verification of SSL certificates during requests made by commonWP:

```php
add_filter( 'http_request_args', function( $args, $url ) {
	if (
		false !== strpos( $args['user-agent'], 'commonWP' )
		&& 0 !== strpos( $url, 'https://cdn.jsdelivr.net/' )
	) {
		$args['sslverify'] = false;
	}

	return $args;
}, 10, 2 );
```
Let me know if this code helps.

**Comment by brandonkal** [comment]
That is an interesting approach. I ended up adding my devcert root CA to the wp-includes/certificates/ca-bundle.crt file. Since that file is not in my source control, it should be fine for local development.

Yes, thank you for making the WP-CLI commands. 👍

It took some time to debug as I don't have of an elegant way to debug background tasks with xdebug. Ended up adding a bunch of error_log() commands to find out which lines were not being executed.

It is interesting that the remote_get approach is more reliable. So I will just close this now!
